### ISSUE TITLE
Imcompatibility with scipy 1.13.0

### ISSUE THREAD

**Issue by sss441803** [jcmgray/quimb#225]
### What happened?

When pip installing quimb, it automatically uses the latest scipy, which implemented changes w.r.t. to sparse matrices.

### What did you expect to happen?

No error.

### Minimal Complete Verifiable Example

```Python
pip install quimb

```python
import quimb.tensor as qtn
```
```


### Relevant log output

```Python
Traceback (most recent call last):
...omitted for privacy
    import quimb.tensor as qtn
  File ".../quimb/__init__.py", line 26, in <module>
    from .core import (
  File ".../quimb/core.py", line 2186, in <module>
    sp.csr_matrix._mul_vector = csr_mulvec_wrap(sp.csr_matrix._mul_vector)
                                                ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'csr_matrix' has no attribute '_mul_vector'. Did you mean: '_matmul_vector'?
```

### Anything else we need to know?

Reinstalling scipy==1.12.0 resolves the issue. The scipy 1.13.0 was released on April 2.

### Environment

This is pip installed release. Github version also requests scipy>1.0 so should have the same issue.

**Comment by jcmgray** [comment]
This is I think a duplicate of #222, which should be fixed in `main` but not released yet, will mint a release soon..

**Comment by jcmgray** [comment]
I have released v1.8.0 with the fix, so closing for the moment!
